Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -366,15 +366,23 @@ extension HTTPClient {
     /// `EventLoopFuture<Response>` of the execution or cancellation of the execution.
     public final class Task<Response> {
         /// `EventLoop` used to execute and process this request.
-        public let eventLoop: EventLoop
-        let promise: EventLoopPromise<Response>
+        public var currentEventLoop: EventLoop {
+            return self.lock.withLock {
+                _currentEventLoop
+            }
+        }
 
+        /// The stored property used by `currentEventLoop` in combination with the `lock`
+        ///
+        /// In most cases you should use `currentEventLoop` instead
+        var _currentEventLoop: EventLoop
```

Comment:
Hey, is this internal read/write access for this underscored var necessary? Couldn't it be `private`?

Reply:
@t089 Thank you for noticing it. If no one sees the need for a unit test to ever access this directly then I can make it `private`.

Reply:
@t089 / @adtrevor yeah, it should be `private`, the `_` is just because there's a `public var currentEventLoop` but the access to that needs to be synchronised. The actual storage is `_currentEventLoop`

Reply:
Ok, thank you @t089, this is fixed now!